### ISSUE TITLE
libgit2 does not call the proper openssl API to implement SNI

### ISSUE THREAD

**Issue by conet** [libgit2/libgit2#3091]
[SNI](http://en.wikipedia.org/wiki/Server_Name_Indication) is needed to access https servers that are located on endpoints which host multiple virtual servers.

Because ```SSL_set_tlsext_host_name(st->ssl, st->socket->host)``` is not called I get a:

```
SSL error: error:14077410:SSL routines:SSL23_GET_SERVER_HELLO:sslv3 alert handshake failure (-1)
```

When trying to clone a repository from such a server.

**Comment by carlosmn** [comment]
That's right, we don't do anything to support SNI at the moment. If you know how to fix it, we'd gladly look over a PR.

**Comment by carlosmn** [comment]
Fixed via manual merge of #3142
